src/AndcultureCode.CSharp.Sitefinity.Conductors/Domain/DynamicContentRepositoryConductor.cs

```diff
@@ -0,0 +1,126 @@
+
+        #endregion Constants
+
+        #region Public Properties
+
+        /// <summary>
+        /// The provider name used by default to get dynamic content.
+        /// Defaults to "OpenAccessProvider".
+        /// </summary>
+        public string DefaultProviderName { get; set; } = "OpenAccessProvider";
+
+        #endregion Public Properties
+
+        #region Private Properties
+
+        private ILogger<IDynamicContentRepositoryConductor> _logger { get; }
+
+        #endregion Private Properties
+
+        #region Public Methods
+
+        public DynamicContentRepositoryConductor(
+            ILogger<IDynamicContentRepositoryConductor> logger
+        )
+        {
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Gets live content from the dynamic module manager by type, and optionally applies a filter.
+        /// </summary>
+        /// <param name="contentType">String representation of the content type.</param>
+        /// <param name="filter">Optional filter to apply to result set.</param>
+        /// <returns></returns>
+        public IResult<IQueryable<DynamicContent>> GetLiveContentByType(
+            string contentType,
+            Expression<Func<DynamicContent, bool>> filter = null
+        ) => GetContentByType(contentType, ContentLifecycleStatus.Live, filter);
+
+        /// <summary>
+        /// Gets content from the dynamic module manager by type, and optionally applies a filter.
+        /// </summary>
+        /// <param name="contentType">String representation of the content type.</param>
+        /// <param name="status">Status of the content you want returned. Null returns content in all statuses.</param>
+        /// <param name="filter">Optional filter to apply to result set.</param>
+        /// <returns></returns>
+        public IResult<IQueryable<DynamicContent>> GetContentByType(
+            string contentType,
+            ContentLifecycleStatus? status = null,
+            Expression<Func<DynamicContent, bool>> filter = null
+        ) => Do<IQueryable<DynamicContent>>.Try((r) =>
+        {
+            if (string.IsNullOrWhiteSpace(contentType))
+            {
+                r.AddErrorAndLog(
+                    _logger,
+                    ERROR_CONTENT_TYPE_NULL_OR_EMPTY,
+                    $"Content type passed into method {nameof(GetContentByType)} is null or empty."
+                );
+                return null;
+            }
+
+            var providerName = DefaultProviderName;
+
+            DynamicModuleManager dynamicModuleManager = DynamicModuleManager.GetManager(providerName);
+
+            if (dynamicModuleManager == null)
+            {
+                r.AddErrorAndLog(
+                    _logger,
+                    ERROR_FAILED_TO_GET_DYNAMIC_MODULE_MANAGER,
+                    $"Attempt to get dynamic module manager with provider name {providerName} in method {nameof(GetContentByType)} returned null or empty."
```

Comment:
Question: Should this error message include `or empty` considering the expression above only checks for the object to be null?